src/octree/node.rs

```
@@ -133,6 +142,37 @@ impl NodeId {
     pub fn index(&self) -> usize {
         self.index as usize
     }
+
+    /// Computes the bounding cube from a NodeID.
+    pub fn bounding_cube(&self, root_bounding_cube: &Cube) -> Cube {
+        let mut edge_length = root_bounding_cube.edge_length();
+        let mut min = root_bounding_cube.min();
+        for level in (0..self.level).rev() {
+            edge_length /= 2.;
+            // Reverse order: process from root to leaf nodes.
+            let child_index = (self.index >> (3 * level)) & 7;
+            let z = (child_index >> 0) & 1;
+            let y = (child_index >> 1) & 1;
+            let x = (child_index >> 2) & 1;
+            min.x += x as f32 * edge_length;
+            min.y += y as f32 * edge_length;
+            min.z += z as f32 * edge_length;
+        }
+        Cube::new(min, edge_length)
+    }
+
+    // Get number of points from the file size of the color data.
+    // Color data is required and always present.
+    fn number_of_points(&self, directory: &Path) -> Result<i64> {
+        let file_meta_data_opt = fs::metadata(self.get_stem(directory).with_extension(COLOR_EXT));
+        if file_meta_data_opt.is_err() {
+            return Err(ErrorKind::NodeNotFound.into())
+        }
+
+        let file_size_bytes = file_meta_data_opt.unwrap().len();
```

In which cases can unwrap fail here? if they are reasonbale, we should probably return an error. If they unreasonable, please add a comment why.

---

From the documentation, errors occurs when:
1. The user lacks permissions to perform metadata call on path.
2. path does not exist.
We return NodeNotFound in any case. Do you think we need more diverse return values?

---

I think that is fine.